src/components/ContentHighlights/HighlightStepper/HighlightStepperTitleInput.jsx

```diff
@@ -11,37 +11,59 @@ const HighlightStepperTitleInput = () => {
   const highlightTitle = useContextSelector(ContentHighlightsContext, v => v[0].stepperModal.highlightTitle);
   const [titleLength, setTitleLength] = useState(highlightTitle?.length || 0);
   const [isInvalid, setIsInvalid] = useState(false);
-
-  const handleChange = (e) => {
+  const [highlightValue, setHighlightValue] = useState({
+    initialized: false,
+    highlightTitle: highlightTitle || '',
+    titleStepValidationError: undefined,
+    highlightTitleLength: titleLength || 0,
+  });
```

Comment:
[curious/clarification] I'm a bit confused as to why we need to keep track of `titleLength` and `highlightTitle` in 2 separate state variables? In `titleLength` (`highlightValue.highlightTitleLength`) and `highlightTitle` (`highlightValue.highlightTitle`).

Do you mind clarifying this and/or adding some documentation as a comment to explain this logic?

Reply:
Yep, in order to separate the rendering from the internal component and external stepper, the internal input needed its own unique state as to not re-render the target value state. I will add additional documentation in reference to this.

Reply:
Got it, cool. Yes, this explanation helps me better understand the changes here :)